### ISSUE TITLE
Backslash get stripped from the sql file when using gzip

### ISSUE THREAD

**Issue by kainxspirits** [spatie/laravel-backup#922]
When using the gzip compressor, the escape character (backslash) got stripped from the file.

I tried to manually run the mysqldump & gzip commands but couldn't reproduce the issue. But it happens each time I'm using the package though.

Here is an example of the generated script: 

```
if output=$('/usr/bin/mysqldump' --defaults-extra-file="/tmp/phpJ1Zk9D" --skip-comments --extended-insert --single-transaction --ignore-table=mydb.telescope_entries --default-character-set=utf8mb4 --hex-blob mydb);\n
then\n
  echo "$output" | gzip > "/var/www/foo/releases/1/storage/app/backup-temp/temp/db-dumps/mysql-foo.sql.gz"\n
else\n
  echo "Dump was not succesful." >&2\n
  exit 1\n
fi
```

I'm using version 6.0.7 with Laravel 5.7 and PHP 7.3.

Thanks for the great package :)

Edit : It might be `echo` that may have a different behavior when used inside `system()` (I believe this is what is used for running the command)

**Comment by rubenvanassche** [comment]
Hi @kainxspirits, i'm looking right now into this issue but cannot reproduce it. So first i tried to run the script manually in bash and indeed it did work as expected. 

So then I wrote a test to check what happens if we run the script inside the Symphony Process component. This is the method used by DBDumper itself for running commands. The test also worked fine, the dumped file had still all the backslashes. 

Can you please try to run this test? Maybe the results differ.

```php
    /** @test */
    public function it_does_not_work()
    {
        $command = <<<BASH
if output=\$(mysqldump --skip-comments --extended-insert --single-transaction --default-character-set=utf8mb4 --hex-blob --user=root bold);
then
  echo "\$output" | gzip > dump.sql.gz
  gzip -d -f dump.sql.gz 
else
  echo "Dump was not succesful." >&2
  exit 1
fi
BASH;

        $process = Process::fromShellCommandline($command, null, null, null,100);

        $process->run();
    }
```

**Comment by kainxspirits** [comment]
Hi @rubenvanassche 

I got the same issue with your test. Bash script works, but backslashes get removed when used with Symfony Process. So I believe the issue comes from that component.

I don't know if that can help to reproduce it but i'm using symfony/process 4.3.1 and db-dumper is version 2.14.1. 
The OS is Ubuntu 16.04.

Thanks for you help.

**Comment by rubenvanassche** [comment]
Hi @kainxspirits 

I've just tried to reproduce this issue on a forge provisioned server. This runs a Ubuntu 18.04 with db-dumper 2.14.1 and 4.3.1. Everything looks to work as it should. The change in operating system version(18.04 vs 16.04) shouldn't be a problem I think.  I'm sorry I can't find a solution for your problem. Maybe you can make an issue at the Symphony Process repository? I will close this issue for now.

**Comment by kainxspirits** [comment]
After making a few more research, I found this post explaining the different beahviors of echo : https://unix.stackexchange.com/a/65819.

So the best thing to improve the reliability of this package across the different shells would be to use `printf "%s\n" "\$output"`. I tested it, and it works in my case. Since it's a pretty simple bash script, that should be ok in most case.

Let me know your thoughts about this change.
